### ISSUE TITLE
Jetty 12 default error handler throws IllegalStateException for application/json

### ISSUE THREAD

**Issue by uhrm** [eclipse/jetty.project#10474]
**Jetty version(s)**
Tested with Jetty 12.0.1

**Jetty Environment**
Jetty 12, EE10

**Java version/vendor**
```
openjdk 19 2022-09-20
OpenJDK Runtime Environment Zulu19.28+81-CA (build 19+36)
OpenJDK 64-Bit Server VM Zulu19.28+81-CA (build 19+36, mixed mode, sharing)
```

**OS type/version**
Windows 11

**Description**
The default `org.eclipse.jetty.server.handler.ErrorHandler` throws an IllegalStateException for content type `application/json`.

The reason is that a JSON return type is supported for both `text/json` and `application/json`, but the JSON error response is only written for `text/json`:

```java
switch (type)
{
    case TEXT_HTML -> writeErrorHtml(request, writer, charset, code, message, cause, showStacks);
    case TEXT_JSON -> writeErrorJson(request, writer, code, message, cause, showStacks);
    case TEXT_PLAIN -> writeErrorPlain(request, writer, code, message, cause, showStacks);
    default -> throw new IllegalStateException();
}
```
(see [ErrorHandler.java, lines 223ff](https://github.com/eclipse/jetty.project/blob/476874584943daf3041eb0ca584d4de3a87f19fd/jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ErrorHandler.java#L223)).

This code should also handle `application/json`:

```java
switch (type)
{
    case TEXT_HTML -> writeErrorHtml(request, writer, charset, code, message, cause, showStacks);
    case TEXT_JSON, APPLICATION_JSON ->
        writeErrorJson(request, writer, code, message, cause, showStacks);
    case TEXT_PLAIN -> writeErrorPlain(request, writer, code, message, cause, showStacks);
    default -> throw new IllegalStateException();
}
```

**How to reproduce?**
Make an HTTP request that triggers the error handler with an `Accept: application/json` header.

**Comment by joakime** [comment]
We should support `application/json` if we currently are not.

The IANA registered MIME media-type is `application/json` after all.

* https://datatracker.ietf.org/doc/html/rfc7159#section-11

**Comment by janbartel** [comment]
Fixed via #10504
